support/hololens/ServoApp/ServoControl/ServoControl.cpp

```diff
@@ -89,6 +89,48 @@ void ServoControl::OnLoaded(IInspectable const &, RoutedEventArgs const &) {
   StartRenderLoop();
 }
 
+void ServoControl::InitializeTextController() {
+  mInputPane = Windows::UI::ViewManagement::InputPane::GetForCurrentView();
+  auto manager = CoreTextServicesManager::GetForCurrentView();
+  mEditContext = manager.CreateEditContext();
+  mEditContext->InputPaneDisplayPolicy(CoreTextInputPaneDisplayPolicy::Manual);
+  mEditContext->InputScope(CoreTextInputScope::Text);
+  mEditContext->TextRequested([=](const auto &, const auto &e) {
+    e.Request().Text(L"");
+    CoreTextRange sel;
+    sel.StartCaretPosition = 0;
+    sel.EndCaretPosition = 0;
+    e.Request().Range() = sel;
+  });
+
+  mEditContext->TextUpdating([=](const auto &, const auto &e) {
+    auto keystr = *hstring2char((e.Text()));
+    RunOnGLThread([=] { mServo->KeyDown(keystr); });
+  });
+
+  mEditContext->SelectionRequested([](const auto &, const auto &) {});
+
+  GotFocus(
+      [=](const auto &, const auto &) { mEditContext->NotifyFocusEnter(); });
+  LostFocus(
+      [=](const auto &, const auto &) { mEditContext->NotifyFocusLeave(); });
+
+  PreviewKeyDown([=](const auto &, const auto &e) {
+    auto keystr = KeyToString(e.Key());
+    if (keystr.has_value()) {
+      auto keyname = *keystr;
+      RunOnGLThread([=] { mServo->KeyDown(keyname); });
+    }
+  });
+  PreviewKeyUp([=](const auto &, const auto &e) {
+    auto keystr = KeyToString(e.Key());
+    if (keystr.has_value()) {
+      auto keyname = *keystr;
+      RunOnGLThread([=] { mServo->KeyUp(keyname); });
```

Comment:
Same concern about #27115, even though this is a std::optional value. We should dereference keyname inside the closure to avoid references to memory that isn't owned by the closure.